### ISSUE TITLE
Feature request: support for BatchNorm1d variant

### ISSUE THREAD

**Issue by vadimkantorov** [mapillary/inplace_abn#138]
This would be super-useful for speech recognition wav2letter model variants.

Thanks :)

**Comment by vadimkantorov** [comment]
I created a feature request in core pytorch about providing the necessary instruments to simplify implementation of inplace-activated batchnorm. Currently I'm proposing an out argument for F.batch_norm and exposing the backward method that accepts input (that can be recomputed manually).

Please feel free to comment on what else is needed (and if the above are helpful): https://github.com/pytorch/pytorch/issues/26288

**Comment by vadimkantorov** [comment]
oh cool :) i will try it. 

For core pytorch, should that be enough if [native_batch_norm](https://github.com/pytorch/pytorch/blob/3f72bcfcaacadfbbfb40e1cc32fc15fdec26bca5/aten/src/ATen/native/native_functions.yaml#L1917) accepts `output` argument 

[native_batch_norm_backward](https://github.com/pytorch/pytorch/blob/3f72bcfcaacadfbbfb40e1cc32fc15fdec26bca5/aten/src/ATen/native/native_functions.yaml#L1941) seems to already accept `input`

**Comment by vadimkantorov** [comment]
After merging https://github.com/pytorch/pytorch/pull/27621, one can implement InplaceBatchNormLeakyRelu like this: https://github.com/pytorch/pytorch/issues/26288#issuecomment-539251421 Please let me know if this looks reasonable
